### PR TITLE
Use arguments instead of env vars for TP comm overlap

### PR DESCRIPTION
**Pull request by minitu**

### CODE REVIEW

File: transformer_engine/pytorch/transformer.py [hunk 1]
```diff
@@ -193,6 +193,18 @@ class TransformerLayer(torch.nn.Module):
...
+    ub_bulk_wgrad : bool, default = `True`
+    ub_bulk_dgrad : bool, default = `True`
+    ub_split_ag : bool, default = `True`
+                  enables split-pipelined overlap of allgather with computation.
+    ub_split_rs : bool, default = `True`
+                  enables split-pipelined overlap of reduce-scatter with computation.
+    ub_atomic_gemm_ag: bool, default = `False`
+                  if set to `True`, enables atomic overlap of allgather with computation.
+    ub_atomic_gemm_rs: bool, default = `False`
+                  if set to `True`, enables atomic overlap of reduce-scatter with computation.
```
Comment (ksivaman) [inline transformer_engine/pytorch/transformer.py:207]:
```suggestion
```

**Comment by ksivaman** [comment]
Could you sign off your commits as per [this guide](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst#sign-your-work)? @minitu

File: transformer_engine/pytorch/csrc/comm_gemm_overlap.h [hunk 2]
```diff
@@ -95,6 +95,9 @@ struct UbufCommOverlap : torch::CustomClassHolder, UbufBase {
     }
     _ubuf = torch::from_blob(_ubuf_ptr, {sample.size(0), sample.size(1)}, sample.options());
 
+    // FIXME: We no longer rely on NVTE_UB_ATOMIC_GEMM_RS, so we cannot determine
```
Comment (minitu) [inline transformer_engine/pytorch/csrc/comm_gemm_overlap.h:98]: PR is ready for review as it passed testing.
Could you advise on what to do regarding this print code block?
CC @ksivaman

Comment (timmoon10) [reply transformer_engine/pytorch/csrc/comm_gemm_overlap.h:98]: I think it's fine to remove this since it's just debugging messages.

**Review by timmoon10 — APPROVED** [review]
LGTM

File: transformer_engine/pytorch/transformer.py (hunk 1 above)
Comment (timmoon10) [reply transformer_engine/pytorch/transformer.py]: Why did we want to remove this documentation? To remove clutter?

**Comment by minitu** [comment]
Rebased to ToT

**Pull request merged**